contracts/competition/Competition.sol

```diff
@@ -0,0 +1,138 @@
+pragma solidity ^0.6.0;
+
+import "../referrals/interfaces/IReferralStorage.sol";
+
+contract Competition
+{
+    uint public start;
+    uint public end;
+    uint public registrationStart;
+    uint public registrationEnd;
+    IReferralStorage private referralStorage;
+
+    struct Team {
+        address leader;
+        string name;
+        bytes32 referral;
+        address[] members;
+        address[] joinRequests;
+    }
+
+    address[] private leaders;
+    mapping(address => Team) private teams;
+    mapping(string => bool) private teamNames;
+    mapping(address => address) private membersToTeam;
+    mapping(address => mapping(address => bool)) private requests;
+
+    modifier registrationIsOpen()
```

Comment:
we don't have solidity linter unfortunately, could you please check solidity styleguides https://docs.soliditylang.org/en/v0.8.16/style-guide.html, we're trying to follow it